src/main/java/reactor/rabbitmq/QueueSpecification.java

```diff
@@ -139,5 +152,13 @@ public QueueSpecification autoDelete(boolean autoDelete) {
             }
             return this;
         }
+
+        @Override
+        public QueueSpecification passive(boolean passive) {
+            if (this.passive != passive) {
```

Comment:
Should it be `this.name == null && passive`? Otherwise the condition does not seem consistent with the error message.

Reply:
Good point, but does this not mean the checks above also need the additional check? For example in:

```java
@Override
public QueueSpecification exclusive(boolean exclusive) {
     if (this.exclusive != exclusive) {
        throw new IllegalArgumentException("Once a queue has a null name, exclusive is always true");
    }
    return this;
}
```

Reply:
Scratch what I've said, I missed it was in the `NullNameQueueSpecification` inner class.

Reply:
Unless you want to add the tests I mentioned, I can merge the PR as-is.

Reply:
I'll happily add the test in a following commit. Not entirely sure in which cases it should apply though. For example in case the queue is declared as passive by the developer, I believe we should not create it from the library in the first place. So should the check be done in case we send the first message to a non passive queue? Or other cases as well?

Reply:
I was thinking of an example that would answer the typical question "how can I make sure the resources I'm publishing to actually exist", so maybe something like:

```java
sender.declare(exchange(exchangeName).passive(true)) // passive declaration
  .onErrorResume(exception -> sender.declare(exchange(exchangeName))) // declaration if error
  .then(sender.declare(queue(queueName).passive(true))) // passive declaration
  .onErrorResume(exception -> sender.declare(queue(queueName))) // declaration if error
  .then(sender.bind(binding(exchangeName, routingKey, queueName)))
  .thenMany(sender.send(Flux.range(0, nbMessages)
                            .map(i -> new OutboundMessage(exchangeName, routingKey, i.toString().getBytes()))))
...
```

I think this would be the typical usage of passive declaration. Other Reactor error handling methods could work as well.

Reply:
Ah, I misunderstood you, I was thinking you meant a test (as in validation) somewhere in the code, rather than a unit test. I will add it.

I would think another typical usage would be a scenario where the exchange/queue was created up front by the RabbitMQ ops rather than it being created through the code.